sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesWalker.cs

```diff
@@ -81,11 +92,48 @@ private void FindExecutableLines(SyntaxNode node)
 
                 case SyntaxKind.ArrayInitializerExpression:
                     executableLineNumbers.Add(node.GetLocation().GetLineNumberToReport());
-                    return;
+                    return true;
+
+                case SyntaxKind.StructDeclaration:
+                case SyntaxKind.ClassDeclaration:
+                    var typeDeclarationSyntax = (BaseTypeDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(typeDeclarationSyntax.AttributeLists);
+
+                case SyntaxKind.MethodDeclaration:
+                    var methodSyntax = (BaseMethodDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(methodSyntax.AttributeLists);
+
+                case SyntaxKind.PropertyDeclaration:
+                    var propertySyntax = (PropertyDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(propertySyntax.AttributeLists);
+
+                case SyntaxKind.EventDeclaration:
+                    var eventSyntax = (EventDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(eventSyntax.AttributeLists);
 
                 default:
-                    return;
+                    return true;
             }
         }
+
+        private bool HasExcludedCodeAttribute(SyntaxList<AttributeListSyntax> attributeSyntaxLists)
+        {
+            var attributes = attributeSyntaxLists.SelectMany(attributeList => attributeList.Attributes);
+            return attributes
+                .OfType<AttributeSyntax>()
+                .Select(GetAttributeName)
+                .Any(IsExcludedAttribute);
+        }
+
+        private static string GetAttributeName(AttributeSyntax attribute)
+        {
+            return attribute?.Name?.ToString() ?? string.Empty;
+        }
+
+        private static bool IsExcludedAttribute(string attributeName)
+        {
+            return ExcludeFromCoverageAttributes.Any(attr =>
+                attributeName.EndsWith(attr, StringComparison.Ordinal));

```

Comment:
This will allow every attribute ending with `ExcludeFromCodeCoverage` to count here. I am not sure this is what we want.